weed/s3api/auth_credentials.go

```diff
@@ -170,35 +203,91 @@ func (iam *IdentityAccessManagement) isEnabled() bool {
 }
 
 func (iam *IdentityAccessManagement) lookupByAccessKey(accessKey string) (identity *Identity, cred *Credential, found bool) {
-
-	err := iam.syncIdentitiesFromKnox()
-	if err != nil {
-		glog.Warning("could not sync with vault")
-		glog.V(3).Infof("Vault error: %v", err)
-	}
 	iam.m.RLock()
 	defer iam.m.RUnlock()
+	if iam.identities == nil {
+		return nil, nil, false
+	}
 	for _, ident := range iam.identities {
-		glog.V(3).Infof("Identity: %+v", ident)
-		for _, cred := range ident.Credentials {
-			glog.V(3).Infof("\tCredential: %+v", cred)
-			// println("checking", ident.Name, cred.AccessKey)
-			if cred.AccessKey == accessKey {
-				return ident, cred, true
+		if ident != nil && ident.Credentials != nil {
```

Comment:
This part could be a bit confusing. My suggestion is to separate code into small funcs/methods in order to make it easy to understand.